src/analysis/convex/basic.lean

```diff
@@ -599,6 +599,37 @@ begin
   convert this; symmetry; simp only [div_eq_iff (ne_of_gt B), y]; ring
 end
 
+lemma convex_on.slope_mono_adjacent {s : set ℝ} {f : ℝ → ℝ} {x y z : ℝ} (hx : x ∈ s) (hz : z ∈ s)
+  (hxy : x < y) (hyz : y < z) (hf : convex_on s f) : (f y - f x) / (y - x) ≤ (f z - f y) / (z - y) :=
```

Comment:
Could you please add an `iff` lemma combining this with `convex_on_real_of_slope_mono_adjacent`?